serial/rfc2217.py

```diff
@@ -722,6 +722,23 @@ def cd(self):
             raise portNotOpenError
         return bool(self.get_modem_state() & MODEMSTATE_MASK_CD)
 
+    @property
+    def timeout(self):
+        """Get the current timeout setting."""
+        return self._timeout
+
+    @timeout.setter
+    def timeout(self, timeout):
+        """Change timeout setting."""
+        if timeout is not None:
+            try:
+                timeout + 1     # test if it's a number, will throw a TypeError if not...
```

Comment:
why not to use `isinstance(timeout, (int, float))`? (use `six.integer_types` in case of python2 support)

Reply:
This is a complete copy of: https://github.com/pyserial/pyserial/blob/3a6ae5a630589a29af8d8bd45616d7aadb2fb3db/serial/serialutil.py#L358-L370 which does not reconfigure the port and in this case copies the weird typecheck :grin: